
robot_motion:
  ap: {'r1', 'r2', 'r3', 'r4', 'r5', 'r6', 'r', 'b'}
  # +-----+-----+-----+
  # | r4,r| r5,b| r6,b|
  # +-----+-----+-----+
  # | r1,r| r2,b| r3,r|
  # +-----+-----+-----+

  regions: 
    - 
      name: 'r1'
      state: [0, 0, 1]
      label: 'r'
    - 
      name: 'r2'
      state: [1, 0, 1]
      label: 'b' 
    - 
      name: 'r3'
      state: [2, 0, 1]
      label: 'r'
    - 
      name: 'r4'
      state: [0, 1, 1]
      label: 'r'
    - 
      name: 'r5'
      state: [1, 1, 1]
      label: 'b'
    - 
      name: 'r6'
      state: [2, 1, 1]
      label: 'b'

  edges: 
    - ['r1', 'r2']
    - ['r2', 'r3']
    - ['r4', 'r5']     
    - ['r5', 'r6'] 
    - ['r1', 'r4']   
    - ['r2', 'r5']      
    - ['r3', 'r6']    

robot_action:
  action: 
    - 
      name: 'pick'
      cost: 100
      guard_formula: 'r'
    -  
      name: 'drop'
      cost: 50
      guard_formula: 'b'

tasks:
  hard_task: '<>(r1 && <> (r2 && <> r6)) && ([]<> r6) && ([]<> r1) && [] <>(r1 -> X r5)'
  soft_task: ''